src/main.cpp

```diff
@@ -1217,6 +1233,9 @@ int main(int Argc, char *Argv[]) {
                 Warning(SourcePos(), "Emitting spir-v file for Xe targets.");
                 ot = Module::SPIRV;
             }
+            // Disable NSW bit optimization by default due to
+            // performance regressions on some GPU workloads.
+            g->wrapSignedInt = true;
```

Comment:
Splitting logic for enabling it for CPU and disabling it for GPU seems really bad idea. Everything should be in single place.

Reply:
Fixed.

Reply:
You moved GPU logic from being below the main logic to being above main logic. You also changed semantics, so now using the switch on GPU will have no effect. Please do as I requested - just have the logic in single place. The "default" case should be just different for GPU and CPU targets, that's it.

Reply:
> You moved GPU logic from being below the main logic to being above main logic.

The first place where we identify that the target is a GPU is within the for loop here. Even though we only support single module compilation for GPU, we need this loop. Setting this here and then immediately doing the rest after the loop keeps the logic close together.

> You also changed semantics

Fixed.